dsgrid/registry/project_registry_manager.py

```diff
@@ -789,6 +793,61 @@ def register_supplemental_dimensions(
         finally:
             context.finalize(error_occurred)
 
+    def add_dataset_requirements(
+        self,
+        project_id: str,
+        filename: Path,
+        submitter: str,
+        log_message: str,
+    ):
+        """Add requirements for one or more datasets to the project."""
+        config = self.get_by_id(project_id)
+        model = InputDatasetListModel.from_file(filename)
+        existing_ids = {x.dataset_id for x in config.model.datasets}
+        for dataset in model.datasets:
+            if dataset.dataset_id in existing_ids:
+                msg = f"{dataset.dataset_id} is already stored in the project"
+                raise DSGInvalidParameter(msg)
+            if dataset.status != DatasetRegistryStatus.UNREGISTERED:
+                msg = f"New dataset {dataset.dataset_id} status must be unregistered: {dataset.status}"
+                raise DSGInvalidParameter(msg)
+
+        config.model.datasets += model.datasets
+        self._update_config(config, submitter, VersionUpdateType.PATCH, log_message)
+
+    def replace_dataset_dimension_requirements(
+        self,
+        project_id: str,
+        filename: Path,
+        submitter: str,
+        log_message: str,
+    ):
+        """Replace dataset requirements in a project."""
+        config = self.get_by_id(project_id)
+        model = InputDatasetDimensionRequirementsListModel.from_file(filename)
+        for dataset in model.datasets:
+            found = False
+            for i in range(len(config.model.datasets)):
```

Comment:
`For I, dataset in enumerate(config.model.datasets):`

Reply:
I used an index to make it more obvious that we are changing the config as well as to avoid an odd variable name (`dataset` is already used in this scope).